connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1745,6 +1958,23 @@ private boolean checkRebalanceNeeded(Callback<?> callback) {
         return false;
     }
 
+    /**
+     * Execute the given action and subsequent callback immediately if the current thread is the herder's tick thread,
+     * or use them to create and store a {@link DistributedHerderRequest} on the request queue and return the resulting request
+     * if not.
+     * @param action the action that should be run on the herder's tick thread
+     * @param callback the callback that should be invoked once the action is complete
+     * @return a new {@link DistributedHerderRequest} if one has been created and added to the request queue, and {@code null} otherwise
+     */
+    DistributedHerderRequest addOrRunRequest(Callable<Void> action, Callback<Void> callback) {
```

Comment:
I wonder if `runOnTickThread` might be a better name, since it more explicitly describes what it's doing?

Reply:
That works, yeah 👍